src/Main.cpp

```diff
@@ -86,26 +93,28 @@ class TCPMessageDispatcher : public AbstractTask<2 * configMINIMAL_STACK_SIZE> {
     ILogger& m_logger;
 
     void task() override {
-        auto socket = SocketContainer::getHostClientSocket();
-
-        if (socket) {
-            HiveMindHostDeserializer deserializer(socket.value());
-            MessageDispatcher messageDispatcher(MessageHandlerContainer::getBuzzMsgQueue(),
-                                                MessageHandlerContainer::getHostMsgQueue(),
-                                                MessageHandlerContainer::getRemoteMsgQueue(),
-                                                deserializer, BSPContainer::getBSP().getUUId(),
-                                                m_logger);
-
-            while (true) {
-                if (!messageDispatcher.deserializeAndDispatch()) {
-                    m_logger.log(LogLevel::Warn, "Fail to deserialize/dispatch TCP");
+        while (true) {
+            auto socket = SocketContainer::getHostClientSocket();
+            if (socket) {
+                HiveMindHostDeserializer deserializer(socket.value());
+                MessageDispatcher messageDispatcher(MessageHandlerContainer::getBuzzMsgQueue(),
+                                                    MessageHandlerContainer::getHostMsgQueue(),
+                                                    MessageHandlerContainer::getRemoteMsgQueue(),
```

Comment:
Who's the remote? Cellphone and/or ESP?

Reply:
Remote is the ESP, but in this case, we are treating the cellphone (so the uart communication) as remote for the sake of the demo